### PR TITLE
Transaction

### PR DESCRIPTION
**Pull request by piyushandilya**

### CODE REVIEW

File: src/main/java/com/uptech/accounted/bean/Recipient.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+@Entity
+@Table(name = "Recipient")
+@Getter
+@Setter
+@NoArgsConstructor
+@ToString
+@Builder
+public class Recipient implements Master {
+	@Id
+	@Column(name = "code", updatable = false, nullable = false)
```
Comment (amanmanocha) [inline src/main/java/com/uptech/accounted/bean/Recipient.java]: Formatting?

File: src/main/java/com/uptech/accounted/bean/Ledger.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  @NonNull
+  private String subLedgerCode;
+  @Column(name = "subLedgerName", updatable = true, nullable = false)
+  @NonNull
+  private String subLedgerName;
+
+  public Ledger(long ledgerId, String ledgerCode, String ledgerName, String subLedgerCode, String subLedgerName) {
+    if (StringUtils.isEmpty(ledgerName) || StringUtils.isEmpty(ledgerCode) || StringUtils.isEmpty(subLedgerName)
+        || StringUtils.isEmpty(subLedgerCode)) {
+      validationAlert();
```
Comment (amanmanocha) [inline src/main/java/com/uptech/accounted/bean/Ledger.java]: validationAlert here?

Comment (amanmanocha) [reply src/main/java/com/uptech/accounted/bean/Ledger.java]: Ledger was a different branch?

File: src/main/java/com/uptech/accounted/bean/Transaction.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  @OneToOne
+  @OnDelete(action = OnDeleteAction.CASCADE)
+  private Department department;
+
+  @NonNull
+  private LocalDate dateOfTransaction;
+
+  @NonNull
+  @OneToOne
+  @OnDelete(action = OnDeleteAction.CASCADE)
```
Comment (amanmanocha) [inline src/main/java/com/uptech/accounted/bean/Transaction.java]: Cascade?

File: src/main/java/com/uptech/accounted/controller/MenuBarController.java [hunk 4]
```diff
@@ -39,8 +39,20 @@ public void showInitiatorForm(ActionEvent actionEvent) {
     stageManager.switchScene(FxmlView.INITIATOR);
   }
 
+  public void showLedgerForm(ActionEvent actionEvent) {
```
Comment (amanmanocha) [inline src/main/java/com/uptech/accounted/controller/MenuBarController.java]: Formatting?

File: src/main/java/com/uptech/accounted/controller/TransactionController.java [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+import javafx.scene.control.cell.PropertyValueFactory;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.util.Callback;
+
+@Controller
+public class TransactionController implements Initializable {
+
+	@FXML
+	private ComboBox<String> cbInitiator;
```
Comment (amanmanocha) [inline src/main/java/com/uptech/accounted/controller/TransactionController.java]: Formatting?

File: src/main/java/com/uptech/accounted/controller/TransactionController.java [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+
+	@FXML
+	private Button reset;
+
+	@FXML
+	private Button saveTransaction;
+
+	Callback<TableColumn<Transaction, Boolean>, TableCell<Transaction, Boolean>> cellFactory = new Callback<TableColumn<Transaction, Boolean>, TableCell<Transaction, Boolean>>() {
+		@Override
+		public TableCell<Transaction, Boolean> call(final TableColumn<Transaction, Boolean> param) {
```
Comment (amanmanocha) [inline src/main/java/com/uptech/accounted/controller/TransactionController.java]: What is all this?

File: src/main/java/com/uptech/accounted/controller/TransactionController.java [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+	public String getAmount() {
+		return amount.getText();
+	}
+
+	public String getSubjectMatter() {
+		return subjectMatter.getText();
+	}
+
+	private void loadTransactionDetails() {
+		transactionList.clear();
```
Comment (amanmanocha) [inline src/main/java/com/uptech/accounted/controller/TransactionController.java]: Formatting?

File: src/main/java/com/uptech/accounted/service/TransactionServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+public class TransactionServiceImpl {
+
+  private final TransactionRepository repository;
+
+  public TransactionServiceImpl(@Autowired TransactionRepository repository) {
+    this.repository = repository;
+  }
+
+  public List<Transaction> findAll() {
+	  return repository.findAll();
```
Comment (amanmanocha) [inline src/main/java/com/uptech/accounted/service/TransactionServiceImpl.java]: Formatting?

File: src/main/java/com/uptech/accounted/service/UserServiceImpl.java [hunk 9]
```diff
@@ -16,9 +18,14 @@ public UserServiceImpl(@Autowired UserRepository repository) {
   }
 
   public boolean authenticate(String email, String password) {
-    return repository.findByEmailAndPassword(email, password) != null;
+//    return repository.findByEmailAndPassword(email, password) != null;
```
Comment (amanmanocha) [inline src/main/java/com/uptech/accounted/service/UserServiceImpl.java]: ?

File: src/main/java/com/uptech/accounted/service/UserServiceImpl.java [hunk 10]
```diff
@@ -16,9 +18,14 @@ public UserServiceImpl(@Autowired UserRepository repository) {
...
+	  return true;
   }
 
+  public List<User> findAll() {
+	  return repository.findAll();
```
Comment (amanmanocha) [inline src/main/java/com/uptech/accounted/service/UserServiceImpl.java]: Formatting?

File: src/main/resources/fxml/MenuBar.fxml [hunk 11]
```diff
@@ -1,35 +1,39 @@
...
+            </items>
+        </Menu>
+        <!--<Menu mnemonicParsing="false" text="Edit">-->
+            <!--<items>-->
+                <!--<MenuItem mnemonicParsing="false" onAction="#delete" text="Delete" />-->
+            <!--</items>-->
+        <!--</Menu>-->
+        <Menu mnemonicParsing="false" text="Forms">
+            <items>
+                <MenuItem mnemonicParsing="false" onAction="#showDepartmentForm" text="Department" />
```
Comment (amanmanocha) [inline src/main/resources/fxml/MenuBar.fxml]: Formating?

File: src/main/resources/fxml/MenuBar.fxml [hunk 12]
```diff
@@ -1,35 +1,39 @@
...
-	</menus>
+
+<MenuBar BorderPane.alignment="TOP_LEFT" xmlns="http://javafx.com/javafx/8" xmlns:fx="http://javafx.com/fxml/1" fx:controller="com.uptech.accounted.controller.MenuBarController">
+    <menus>
+        <Menu mnemonicParsing="false" text="File">
+            <items>
+                <MenuItem mnemonicParsing="false" onAction="#exit" text="Logout" />
+            </items>
+        </Menu>
+        <!--<Menu mnemonicParsing="false" text="Edit">-->
```
Comment (amanmanocha) [inline src/main/resources/fxml/MenuBar.fxml]: ?

File: src/main/java/com/uptech/accounted/controller/TransactionController.java (hunk 6 above)
Comment (piyushandilya) [reply src/main/java/com/uptech/accounted/controller/TransactionController.java]: cellFactory to support editing of cells

File: src/main/java/com/uptech/accounted/controller/TransactionController.java [hunk 13]
```diff
@@ -0,0 +1,288 @@
...
+  private void loadDepartments() {
+    departmentComboList.clear();
+    List<Department> departmentList = departmentRepository.findAll();
+    for (Department department : departmentList) {
+      departmentComboList.add(department.getCode() + "-" + department.getName());
+    }
+    cbDepartment.setItems(departmentComboList);
+  }
+
+  private void loadLedgers() {
```
Comment (amanmanocha) [inline src/main/java/com/uptech/accounted/controller/TransactionController.java]: The two methods seem duplicate.

**Pull request merged**